### PR TITLE
Deprecating the offline session preloading

### LINKED ISSUE #25300: Deprecate offline session preloading
### Description

The new default of Keycloak load offline sessions on demand, and not loading them at startup any more. The old code to preload them at startup should be removed. 

### Discussion

https://github.com/keycloak/keycloak/discussions/25301

### Motivation

The old default of preloading is still available and tested. This makes the code unnecessarily complex and more difficult to test. With a lot of offline token it also doesn't scale. 

We also want eventually use the table structure and lookup functionality to work for all sessions, so this would be a training execerise.

### Details

This issue is about checking that not using preloading works as expected in call cases, and makes preload deprecated for the next release so it can be eventually removed.

One known issue that is not supported without preloading is 

https://github.com/keycloak/keycloak/blob/a45934a762d8726d1cd3ea9c2c0e34e09dd5b849/model/infinispan/src/main/java/org/keycloak/models/sessions/infinispan/InfinispanUserSessionProvider.java#L387-L393

### PR DESCRIPTION
**Pull request by ahus1**
Closes #25300

### CODE REVIEW

**Review by pedroigor — DISMISSED** [review]
Finally :)

**Review by vramik — APPROVED** [review]
Changes LGTM, thank you @ahus1

File: testsuite/model/test-all-profiles.sh [hunk 1]
```diff
@@ -11,11 +11,11 @@ mvn -version
 
 EXIT_CODE=0
 mvn clean
-for I in `perl -ne 'print "$1\n" if (m,<id>([^.<]+)</id>,)' pom.xml`; do
+for I in `perl -ne 'print "$1\n" if (m,<id>([^.<]+)</id>,)' pom.xml | grep offline`; do
```
Comment (martin-kanis) [inline testsuite/model/test-all-profiles.sh:14]: I guess, this change is not desired?

File: docs/documentation/server_admin/topics/sessions/preloading.adoc [hunk 2]
```diff
@@ -9,10 +9,11 @@ Therefore, the offline sessions are lazily fetched from the database by default.
...
 However, {project_name} can be configured to preload the offline sessions from the database into the Infinispan caches during the server startup.
 It can be achieved by setting `preloadOfflineSessionsFromDatabase` property in the `userSessions` SPI to `true`.
+This functionality is currently deprecated and will be removed in a future release.
 
 The following example shows how to configure offline sessions preloading.
 
 [source,bash]
 ----
-bin/kc.[sh|bat] start --spi-user-sessions-infinispan-preload-offline-sessions-from-database=true
+bin/kc.[sh|bat] start --features-enabled --spi-user-sessions-infinispan-preload-offline-sessions-from-database=true
```
Comment (sschu) [inline docs/documentation/server_admin/topics/sessions/preloading.adoc:18]: Doesnt this miss the "offline-session-preloading" after the "--features-enabled"?

**Review by martin-kanis — CHANGES_REQUESTED** [review]
Thanks for the PR. I have two comments, see below:

File: docs/documentation/server_admin/topics/sessions/preloading.adoc (hunk 2 above)
Comment (ahus1) [reply docs/documentation/server_admin/topics/sessions/preloading.adoc:18]: Thank you for spotting this, see martin's comment below.

File: testsuite/model/test-all-profiles.sh (hunk 1 above)
Comment (ahus1) [reply testsuite/model/test-all-profiles.sh]: This was unintended, thank you for spotting this.

**Review by martin-kanis — APPROVED** [review]
Looks good, thanks.

**Comment by ahus1** [comment]
@mhajas - with the review in, could you please approve and merge it? Thanks!

**Review by mhajas — APPROVED** [review]
Thank you for this @ahus1! The changes look good to me. I am merging this now but  I wanted to ask in which version do we plan to remove this completely? 25? 26? Should we create a follow-up issue to remove the code for that milestone?

**Pull request merged**